shared/src/balancer/event_handler.rs

```diff
@@ -155,7 +169,30 @@ impl PoolSpecialization {
     }
 }
 
-impl BalancerPools {
+impl PoolRegistry {
+    // Since all the fields are private, we expose helper methods to fetch relevant information
+    pub fn pools_containing_pair(&self, token_pair: TokenPair) -> HashSet<RegisteredPool> {
```

Comment:
I have decided to return HashSet here since All fields in `RegisteredPool` are hashable and this makes the unit tests more deterministic on this function's output.